module/VuFind/src/VuFind/Content/Covers/Bokinfo.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+/**
+ * Plugin for Bokinfo coverimages
+ *
+ * PHP version 7
+ *
+ * Copyright (C) imCode Partner AB 2022.
+ * Copyright (C) Moravian Library 2019.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Jacob Sandin <jacob@imcode.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Content\Covers;
+
+use SimpleXMLElement;
+
+/**
+ * Summon cover Bokinfo content loader.
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Jacob Sandin <jacob@imcode.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Bokinfo extends \VuFind\Content\AbstractCover implements
+    \Laminas\Log\LoggerAwareInterface,
+    \VuFindHttp\HttpServiceAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->cacheAllowed = true;
+        $this->supportsIsbn = true;
+    }
+
+    /**
+     * Get image URL for a particular API key and set of IDs (or false if invalid).
+     *
+     * @param string $key  API key
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        if (!isset($ids['isbn'])) {
+            return false;
+        }
+        if (!isset($key)) {
```

Comment:
Maybe this should be `if (empty($key))` rather than `if (!isset($key))`. By virtue of being a required function parameter, `$key` is always going to be set to something... but if it's an empty string, that's going to be a problem. :-)